ExternalSampleProjects/ExternalSamplesUtility.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Diagnostics;
+using UnityEngine.InputSystem;
+
+public class ExternalSamplesUtility
+{
+	static string token => System.Environment.GetEnvironmentVariable("GITHUB_AUTH");
+
+	struct ReleaseResponse
+	{
+		public string upload_url;
+	}
+	
+	static string CallGithubAPI(string commandUrl, string contentType, string requestData = null)
+	{
+	    var httpWebRequest = (HttpWebRequest)WebRequest.Create(commandUrl);
+        httpWebRequest.ContentType = contentType;
+        httpWebRequest.Method = requestData != null ? "POST" : "GET";
+        httpWebRequest.Accept = "application/vnd.github.v3+json";
+        httpWebRequest.Headers["Authorization"] = $"token {token}";
+        httpWebRequest.UserAgent = "Unity";
+        httpWebRequest.Timeout = 1000000;
+        
+        if (requestData != null)
+        {
+			var stream = httpWebRequest.GetRequestStream();
+			if (contentType == "application/binary")
+			{
+				var data = File.ReadAllBytes(requestData);
+				stream.Write(data, 0, data.Length);
+				stream.Close();
+			}
+			else
+			{
+				using (var streamWriter = new StreamWriter(stream))
+				{
+					streamWriter.Write(requestData);
+				}
+			}
+        }
+
+        try
+        {
+            var httpResponse = (HttpWebResponse) httpWebRequest.GetResponse();
+			using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
+			{
+				var result = streamReader.ReadToEnd();
+				UnityEngine.Debug.Log("Server said " + result);
+				return result;
+			}
+        }
+        catch (WebException e)
+        {
+        	UnityEngine.Debug.Log($"Failed request {commandUrl} with data: '{requestData}'. {e}");
+            using (var streamReader = new StreamReader(e.Response.GetResponseStream()))
+            {
+                var result = streamReader.ReadToEnd();
+                UnityEngine.Debug.Log("Server said " + result);
+            }
+            throw;
+        }
+	}
+	
+	static void RunGitCommannd(string arguments)
+	{
+	    Process p = new Process();
+        p.StartInfo.UseShellExecute = false;
+        p.StartInfo.RedirectStandardOutput = true;
+        p.StartInfo.FileName = "git";
+		p.StartInfo.Arguments = arguments;
+        p.Start();
+        p.WaitForExit();
+	}
+	
+	static string[] GetAssetsToPublish()
+	{
+		return 
+			Directory.GetFileSystemEntries("Assets").Where(x => !x.Contains("ExternalSamplesUtility"))
+			.Concat(new []{"ProjectSettings/ProjectSettings.asset"})
+			.ToArray();
```

Comment:
We'll probably TagManager from ProjectSettings, too (used by TouchSamples). Wondering if we should just grab everything from ProjectSettings?

Also, shouldn't this grab `Packages/manifest.json` too?

Reply:
We can add more, but I don't want to unnecessarily wipe any user settings, so I just wanted to grab the stuff we actually need. So, if TouchSamples needs TagManager, we can get that  - but then where do I find the TagManager set up for TouchSamples? Because the TagManager currently in our input system project just has the default settings in it.

As for `Packages/manifest.json` - can't grab that - it's not an asset.

Reply:
>where do I find the TagManager set up for TouchSamples? Because the TagManager currently in our input system project just has the default settings in it.

It's still pending on PR #745. Didn't make it further yet. Now what we have a proper way to deal with per-sample project settings, would like to land that after your PR.

>As for Packages/manifest.json - can't grab that - it's not an asset.

Doh, I thought we register that as a constant GUID :/

Reply:
Ok, so let's land it, and then add the TagManager